Helper/OpenInvoice.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2023 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Helper;
+
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Catalog\Helper\Image;
+
+class OpenInvoice
+{
+    /**
+     * @var AbstractHelper
+     */
+    protected $adyenHelper;
+
+    /**
+     * @var CartRepositoryInterface
+     */
+    protected $cartRepository;
+
+    /**
+     * @var ChargedCurrency
+     */
+    protected $chargedCurrency;
+
+    /**
+     * @var Config
+     */
+    protected  $configHelper;
+
+    /**
+     * @var Image
+     */
+    protected $imageHelper;
+
+    public function __construct(
+        Data $adyenHelper,
+        CartRepositoryInterface $cartRepository,
+        ChargedCurrency $chargedCurrency,
+        Config $configHelper,
+        Image $imageHelper
+    ) {
+        $this->adyenHelper = $adyenHelper;
+        $this->cartRepository = $cartRepository;
+        $this->chargedCurrency = $chargedCurrency;
+        $this->configHelper = $configHelper;
+        $this->imageHelper = $imageHelper;
+    }
+
+    public function getOpenInvoiceData($order): array
```

Comment:
that's a big function '>100 lines of code', can u split it to smaller functions to enhance readability?